transmart-rest-api/grails-app/controllers/org/transmartproject/rest/UserQuerySetController.groovy

```diff
@@ -0,0 +1,107 @@
+package org.transmartproject.rest
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.bind.annotation.PathVariable
+import org.transmartproject.core.userquery.UserQuerySetDiff
+import org.transmartproject.core.userquery.UserQuerySetInstance
+import org.transmartproject.core.userquery.UserQuerySetResource
+import org.transmartproject.rest.misc.CurrentUser
+
+import static org.transmartproject.rest.misc.RequestUtils.checkForUnsupportedParams
+
+class UserQuerySetController {
+
+    @Autowired
+    CurrentUser currentUser
+
+    @Autowired
+    UserQuerySetResource userQuerySetResource
+
+    static responseFormats = ['json']
+
+    /**
+     * Scans for changes in results of the stored queries and updates stored sets:
+     * <code>/${apiVersion}/query_sets/scan</code>
+     *
+     * This endpoint should be called after loading, deleting or updating the data.
+     * Only available for administrators.
+     *
+     * @return number of sets that were updated, which is also a number of created querySets
+     */
+    def scan() {
+        Integer result = userQuerySetResource.scan(currentUser)
+        response.status = 201
+        respond([numberOfUpdatedSets: result])
+    }
+
+    /**
+     * Gets a list of query sets related to a specific query:
+     * <code>/${apiVersion}/query_sets/${queryId}</code>
```

Comment:
Could be `/${apiVersion}/queries/${queryId}/query_sets`, as it refers to the same query as in the userquerycontroller.

Reply:
This one refers to a userQuerySetController, 
also the call for diffs is:
` /${apiVersion}/queries/${queryId}/query_sets/diffs`
That is why I decided to use `/query_sets/${queryId}`.

Reply:
The controller does not determine the API call. You can distribute arbitrarily over controllers as you like. In this case fetching query objects in one controller and query sets in another one. But these query sets belong to the queries, so it seems natural to use `/queries/${queryId}` in the url. `/query_sets/${queryId}` suggests that the `queryId` identifies one query set.